urbanairship-react-native/js/UrbanAirship.ts

```diff
@@ -642,12 +642,32 @@ export class UrbanAirship {
    * @param eventType The event type. Either EventType.NotificationResponse, EventType.pushReceived,
    * EventType.Register, EventType.Reistration, EventType.DeepLink, EventType.NotificationOptInStatus,
    * EventType.InboxUpdated, or EventType.ShowInbox.
-   * @param listener The event listner.
+   * @param listener The event listner. Should be a reference to the function passed into addListener.
    */
   static removeListener(eventType: EventType, listener: (...args: any[]) => any) {
     EventEmitter.removeListener(convertEventEnum(eventType), listener);
   }
 
+  /**
+   * Removes a subscription for an Urban Airship event.
+   *
+   * @param subscription The emitter subscription, returned from addListener.
+   */
+  static removeSubscription(subscription: EmitterSubscription) {
```

Comment:
I'm actually not sure we need this after all, since you can call `remove()` on the subscription itself which does the exact same thing. https://github.com/facebook/react-native/blob/master/Libraries/vendor/emitter/_EmitterSubscription.js#L53

That's a pretty typical API for things like subscriptions and disposables. The removeAllListeners method is still useful though since it doesn't require hanging onto any state.

Reply:
Me and Brian discussed the first part actually (It should be in our conversation up above)! `removeSubscription` was already created in UAEventEmitter, and there's some pre-existing code in there that I figured was important to Airship, so I thought it may be worth adding it to Urbanairship.ts. If that extra code in the method isn't needed, then we can remove it and have folks use remove().

What do you think?

Reply:
Sorry to leave you hanging there – after more discussion I think we should take out `removeSubscription` and call it good. Since the subscription itself already has a cancel method it's arguably redundant, and rather than giving a bunch of different options we'd be better off just documenting the plugin better.